### ISSUE TITLE
Docs data structures missing file/format/start/end?

### ISSUE THREAD

**Issue by nathanaeljones** [sourcegraph/srclib#112]
Is full docs parsing supported by any toolchain? 

We'd like to build a docco-style browser as a proof-of-concept integration with srclib, but paths and indicies are missing for each toolchain we try. 

    "Docs": [
     {
      "Path": "commonjs/test/vendor/qunit.js/-/assert/deepEqual",
      "Format": "",
      "Data": "@name deepEqual @function",
      "File": "",
      "Start": 0,
      "End": 0
     },

**Comment by sqs** [comment]
Currently none of the toolchains I know of produce file/start/end information for docstrings. However, what you probably want is the file/start/end of the definition that the documentation pertains to. You can obtain that by finding the corresponding def with the same path as the doc's Path. Does that get you what you want?

I will clarify this in the docs.

**Comment by nathanaeljones** [comment]
Thanks, but we do actually want the docstring indicies, as we have to know which parts to delete from the source code during display. Docco [style browsers](http://nevir.github.io/groc/project.html) extract all comments and place them in a left hand pane, while lining up the corresponding source code on the right hand pane. Our end goal is melding docco-style source browsing with traditional class/structure browsing, so that users can switch from 'generated docs' to annotated (and ref navigable) source and back - easily. And making this fast enough to run on Travis CI OSS plans.

**Comment by sqs** [comment]
Got it. Sounds like a great use case and one that I wish srclib supported. It's something we would like to add to the srclib toolchains we've built, and we'd happily accept contributions. But it's not something we plan to do ourselves in the next ~month.

**Comment by nathanaeljones** [comment]
I'm trying to convince some of the OSS folks at Microsoft that srclib would be a good basis for building their new documentation system on, and that it's worthwhile to add toolchains for their compilers. 

I want to build a working demo to show (and verify) what would be possible through srclib. What toolchain would you recommend I adjust if I want to add these indicies?

**Comment by jelmerdereus** [comment]
@nathanaeljones I'm curious if you have any updates on this issue

**Comment by jelmerdereus** [comment]
@beyang do you want to keep it open as something to accomplish?

**Comment by beyang** [comment]
Yes, keep this open, because this is something we want to support.
